test/application/widgets/theme/custom_text_field_test.dart

```diff
@@ -0,0 +1,101 @@
+import 'package:memo/application/widgets/theme/custom_text_field.dart';
+
+import '../../../utils/widget_pump.dart';
+
+void main() {
+  group('Editing State - ', () {
+    testWidgets('should have the expected text in the controller after typing in the TextField', (tester) async {
+      final controller = TextEditingController();
+      final textField = CustomTextField(controller: controller);
+      const expectedText = 'any';
+
+      await pumpMaterialScopedWithTheme(tester, textField);
+      await tester.enterText(find.byType(CustomTextField), expectedText);
+
+      expect(controller.text, expectedText);
+    });
+
+    testWidgets('should have focus when tapping in the TextField', (tester) async {
+      final focusNode = FocusNode();
+      final textField = CustomTextField(focusNode: focusNode);
+
+      await pumpMaterialScopedWithTheme(tester, textField);
+      expect(focusNode.hasFocus, false);
+
+      await tester.tapAt(Offset.zero);
+      expect(focusNode.hasFocus, true);
+    });
+
+    testWidgets('should show clear button when has text and showsClearIcon is true', (tester) async {
+      // ignore: avoid_redundant_argument_values
+      const textField = CustomTextField(showsClearIcon: true);
+
+      await pumpMaterialScopedWithTheme(tester, textField);
+      var renderedTextField = find.byType(TextField).first.evaluate().single.widget as TextField;
+      expect(renderedTextField.decoration!.suffixIcon, null);
+
+      await tester.enterText(find.byType(CustomTextField), 'any');
+      await tester.pumpAndSettle();
+
+      renderedTextField = find.byType(TextField).first.evaluate().single.widget as TextField;
+      expect(renderedTextField.decoration!.suffixIcon, isA<IconButton>());
+    });
+
+    testWidgets('should show custom suffix icon when not focused', (tester) async {
+      final suffixIcon = Container(color: Colors.red);
+      final textField = CustomTextField(suffixIcon: suffixIcon);
+
+      await pumpMaterialScopedWithTheme(tester, textField);
+
+      final renderedTextField = find.byType(TextField).first.evaluate().single.widget as TextField;
+      expect(renderedTextField.decoration!.suffixIcon, suffixIcon);
+    });
+
+    testWidgets('should replace custom suffix icon by clear button when editing', (tester) async {
+      final suffixIcon = Container(color: Colors.red);
+      final textField = CustomTextField(suffixIcon: suffixIcon);
+
+      await pumpMaterialScopedWithTheme(tester, textField);
+      var renderedTextField = find.byType(TextField).first.evaluate().single.widget as TextField;
+      expect(renderedTextField.decoration!.suffixIcon, suffixIcon);
+
+      await tester.enterText(find.byType(CustomTextField), 'any');
+      await tester.pumpAndSettle();
+
+      renderedTextField = find.byType(TextField).first.evaluate().single.widget as TextField;
+      expect(renderedTextField.decoration!.suffixIcon, isA<IconButton>());
+    });
+  });
+
+  group('Error State - ', () {
```

Comment:
"Labels"

What about testing helperText as well here, and while at it, maybe test if both are supplied and who takes precedence?

Reply:
Awesome suggestion